contracts/DepositContract.sol

```diff
@@ -0,0 +1,75 @@
+pragma solidity ^0.4.22;
+
+contract DepositContract {
+
+  mapping(address => uint) private Stakes;
+  mapping(address => uint) private Indices;
+  address[] private nodeAddressList;
+
+  event LogNewNode(address indexed nodeAddress, uint index, uint stake);
+  event LogUpdateStake(address indexed nodeAddress, uint index,  uint stake);
+  event LogDeleteNode(address indexed nodeAddress, uint index);
+  
+  function isNodeAddress(address nodeAddress) public constant returns(bool isIndeed) 
+  {
+    if(nodeAddressList.length == 0) return false;
+    return (nodeAddressList[Indices[nodeAddress]] == nodeAddress);
+  }
+
+function deposit() public payable returns(uint index)
+  {
+    if(isNode(msg.sender)) {
```

Comment:
definitely this code is not working - you can not submit such code

Reply:
Forgot to move all changes from remix. Will fix with next commit.

Reply:
Added now.